scripts/relatedBills.py

```diff
@@ -36,38 +36,68 @@ def loadTitlesIndex(titleIndexPath=PATH_TO_TITLES_INDEX, zip=True):
 
     return titlesIndex
 
+def getSimilarTitles(titlesIndex: dict, same=True):
+    billsRelatedByTitle = {}
+    for title, bills in titlesIndex.items():
+        for bill_outer in bills:
+            similarTitles = billsRelatedByTitle.get(bill_outer)
+            
+            # Initialize the key-value for the bill
+            if not similarTitles:
+                billsRelatedByTitle[bill_outer] = []
+
+            for bill_inner in bills:
+                # Find a matching item, if any, in the list billsRelatedByTitle[bill_outer]
+                # See https://stackoverflow.com/a/1701404/628748
+                bill_index = next((i for i,v in enumerate(billsRelatedByTitle[bill_outer]) if (bill_inner == v.get('billCongressTypeNumber'))), None)
+                if not bill_index:
+                    billsRelatedByTitle[bill_outer].append({
+                        'billCongressTypeNumber': bill_inner,
+                        'titles': [title]
+                    })
+                else:
+                    billsRelatedByTitle[bill_outer][bill_index]['titles'].append(title)
+
+    return billsRelatedByTitle 
 
 def getRelatedBills():
     billsMeta = loadBillsMeta()
     titlesIndex = loadTitlesIndex()
-    billsRelatedIndex = {key: [] for key in billsMeta.keys()}
-    getSameTitles(titlesIndex, billsRelatedIndex)
+    billsRelatedIndex = getSimilarTitles(titlesIndex)
     return billsRelatedIndex
-    # for key, value in billsRelatedIndex.items():
-    #     if len(value) > 1:
-    #         print(key, value)
 
+def makeAndSaveRelatedBills():
+    relatedBills = getRelatedBills()
+    saveBillsMeta(billsMeta=relatedBills,
+                   metaPath=PATH_TO_RELATEDBILLS)
 
-def getSameTitles(titlesIndex, billsRelatedIndex):
-    for title, bills in titlesIndex.items():
-        for billNum in bills:
-            similarList = billsRelatedIndex.get(billNum)
-            if len(similarList) == 0:
-                for bill in bills:
-                    similarList.append({
-                        'billCongressTypeNumber': bill,
-                        'titles': [title]
-                    })
-            else:
-                for item in similarList:
-                    item.get('titles').append(title)
+def main(args, loglevel):
```

Comment:
Added back the 'main' function to run from command-line. Takes output and saves to a file for easy inspection afterward.